GNOME #315437; Debian #416863; Gentoo #156475; fixes building of packages
using glib against GCC 4.2 and 4.3

Index: glib2.0-2.13.4/glib/gutils.h
===================================================================
--- glib2.0-2.13.4.orig/glib/gutils.h	2007-06-16 18:27:37.000000000 +0200
+++ glib2.0-2.13.4/glib/gutils.h	2007-06-16 18:30:08.000000000 +0200
@@ -96,9 +96,14 @@
 #ifdef G_IMPLEMENT_INLINES
 #  define G_INLINE_FUNC
 #  undef  G_CAN_INLINE
-#elif defined (__GNUC__) 
-#  define G_INLINE_FUNC extern inline
-#elif defined (G_CAN_INLINE) 
+#elif defined (__GNUC__)
+#  if __GNUC_PREREQ (4,2) && defined (__STDC_VERSION__) \
+   && __STDC_VERSION__ >= 199901L
+#    define G_INLINE_FUNC extern __inline __attribute__ ((__gnu_inline__))
+#  else
+#    define G_INLINE_FUNC extern __inline
+#  endif
+#elif defined (G_CAN_INLINE)
 #  define G_INLINE_FUNC static inline
 #else /* can't inline */
 #  define G_INLINE_FUNC
